### ISSUE TITLE
Valid JSON error 

### ISSUE THREAD

**Issue by BlaisePx** [erohtar/redditSaver#5]
Hi! Getting this script error:

~/.../Termux/redditSaver $ node redditSaver.js
undefined:1
<!doctype html>
^

SyntaxError: Unexpected token '<', "<!doctype "... is not valid JSON
    at JSON.parse (<anonymous>)
    at createNotes (/storage/emulated/0/Termux/redditSaver/redditSaver.js:58:19)
    at Array.<anonymous> (/storage/emulated/0/Termux/redditSaver/redditSaver.js:33:4)
    at finish (node:internal/streams/writable:747:25)
    at finishMaybe (node:internal/streams/writable:735:9)
    at afterWrite (node:internal/streams/writable:507:3)
    at onwrite (node:internal/streams/writable:480:7)
    at node:internal/fs/streams:410:5
    at FSReqCallback.wrapper [as oncomplete] (node:fs:834:5)

Node.js v19.9.0

Any idea of what the problem could be?

Thanks so much for a wonderful tool!

edit: link I am using is:

https://www.reddit.com/user/RutherfordTheButler/saved.json?feed=xxxxxxxxxxxx&user=RutherfordTheButler

**Comment by erohtar** [comment]
Hi, It looks like the file being downloaded isn't a standard json file. 
Just to be sure, you did replace the xxxxxxxxxxxxx part with your actual key from the linked page, right?

**Comment by BlaisePx** [comment]
I did! I actually grabbed the link, added user/RutherfordtheButler (you should know that if you do as instructed:

Go to [this url](https://ssl.reddit.com/prefs/feeds/) and copy this link:

you get the wrong link. It is missing the user/YOURUSERNAME part of the link and that does not appear until you open the link itself and then you get the right link.

Anyway, yes, I replaced those numbers with xx for security.

**Comment by erohtar** [comment]
I don't know about adding that part - here's what mine looks like in browser when I hover cursor over that JSON link:
![image](https://github.com/erohtar/redditSaver/assets/5120628/b1e80b51-ea90-4451-a2c6-7d1829282072)

Try to save that json from browser and validate it here: https://jsonlint.com

**Comment by BlaisePx** [comment]
What I mean is when I right click and copy the link on the "your saved links" button, it gives me this: https://www.reddit.com/saved.json?feed=seriesofnumbers&user=RutherfordTheButler

But when I open the link in a browser, it changes to this:

https://www.reddit.com/user/RutherfordTheButler/saved.json?feed=seriesofnumbers&user=RutherfordTheButler

Anyway, the only thing that I can guess is causing my problem is that I am using your script on Android with Termux. I will keep trying to find a workaround. I just want my saves to automatically go to my Obsidian when I am on Android and this seemed like the solution. Thank you for your help.

**Comment by karanrajpal14** [comment]
Were you able to find a fix to this? I tried both links, the right click one and the one that it changes to after clicking there. Neither worked for me unfortunately.

**Comment by erohtar** [comment]
@BlaisePx & @karanrajpal14 
Without your saved.json file, I wasn't able to know what was the problem on your end. But just to have a look, I checked my own, and found Reddit recently started blocking the json downloads without having a browser agent declared (which is absurd, considering you'd likely not be viewing the json in a browser normally, but anyway). So since I could see the problem now, I was able to fix it, and my own stopped saves started working now.
Pls try the latest update.
